### PR TITLE
Enhancement proposal for SRV records

### LINKED ISSUE #525: Enhancement proposal for Support SRV record in Lighthouse
**What would you like to be added**:
Enhancement proposal for Support SRV record in Lighthouse for exported services
**Why is this needed**:
To cover the design aspects

### PR DESCRIPTION
**Pull request by aswinsuryan**
Fixes : submariner-io/lighthouse/issues/525

Signed-off-by: Aswin Surayanarayanan <asuryana@redhat.com>

### CODE REVIEW

**Review by mangelajo — APPROVED** [review]

File: lighthouse/srv-records.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# SRV records in Lighthouse
+
+## Summary
+
+Lighthouse only supports A record, but there are use case where we support for other types records is required
```
Comment (nyechiel) [inline lighthouse/srv-records.md]:
```suggestion
Lighthouse only supports DNS A records at the moment, but there are use case that require support for other types records. This proposal is focused on adding support for Service (SRV) records as defined in RFC 2782.
```

File: lighthouse/srv-records.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+## Proposal
+
+This enhancement proposes to add support for SRV records in Lighthouse. The support will be added for both ClusterIP Service
+and Headless Service. The proposal only covers the use case where the service ports have the same name protocol and port number across the exported
+services with the same name and namespace.
+
+## Design Details
+
+The SRV records of the below formats shall be supported.
```
Comment (nyechiel) [inline lighthouse/srv-records.md]:
```suggestion
The SRV records of the below formats shall be supported:
```

File: lighthouse/srv-records.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+### Cluster IP Service
+
+- \<portname\>.\<protocol\>.\<svc-name\>.\<svc-namespace\>.svc.clusterset.local -> will return the SRV record with the port
+number and domain name with matching port name and protocol from one of the joined clusters where the service is available.
+The clusters will be selected based on the Round Robin algorithm.
+
+- \<portname\>.\<protocol\>.**\<cluster-name\>**.\<svc-name\>.\<svc-namespace\>.svc.clusterset.local -> will return SRV record
+from the cluster specified, if available with matching portname and protocol
```
Comment (nyechiel) [inline lighthouse/srv-records.md]:
```suggestion
from the cluster specified, if available with matching portname and protocol.
```

File: lighthouse/srv-records.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+If \<port-name\> and \<protocol\> is not specified one SRV record will be returned for each port configured for the service in
```
Comment (nyechiel) [inline lighthouse/srv-records.md]:
```suggestion
If \<port-name\> and \<protocol\> is not specified, one SRV record will be returned for each port configured for the service in
```

File: lighthouse/srv-records.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+the selected/specified cluster.
+
+### Headless Service
+
+- \<portname\>.\<protocol\>.\<svc-name\>.\<svc-namespace\>.svc.clusterset.local -> will return the SRV records  with port number
```
Comment (nyechiel) [inline lighthouse/srv-records.md]:
```suggestion
- \<portname\>.\<protocol\>.\<svc-name\>.\<svc-namespace\>.svc.clusterset.local -> will return the SRV records with port number
```

File: lighthouse/srv-records.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+and domain name for each backend pod with matching portname and protocol in every joined cluster where the service is available.
+
+- \<portname\>.\<protocol\>.**\<cluster-name\>**.\<svc-name\>.\<svc-namespace\>.svc.clusterset.local -> will return the SRV records
+with port number and domain name for each backend pod with matching portname and protocol, in the specified cluster.
+
+If \<port-name\> and \<protocol\> is not specified one SRV record will be returned for each port configured for the service in the selected
```
Comment (nyechiel) [inline lighthouse/srv-records.md]:
```suggestion
If \<port-name\> and \<protocol\> is not specified, one SRV record will be returned for each port configured for the service in the selected
```

**Review by nyechiel — COMMENTED** [review]
Couple of nits, but LGTM. Thanks @aswinsuryan

**Review by nyechiel — APPROVED** [review]

**Pull request merged**